### PR TITLE
WT-13515 Log modularity improvement move files prefix functions move structures

### PR DESCRIPTION
**Pull request by bvpvamsikrishna**
1) moved the src/cursor/cur_log.c --> src/log/cur_log.c

2) external interface --> internal interface  
- wt_log_force_write --> wti_log_force_write

3) internal interface --> external interface 
- wti_cursor_search_near_notsup --> wt_cursor_search_near_notsup
- wti_cursor_modify_notsup --> wt_cursor_modify_notsup

4) moved src/cursor/cur_json.c --> src/support/json.c
5) moved wt_cursor_json as wt_json to new support/json.h

### CODE REVIEW

File: dist/filelist.bzl [hunk 1]
```diff
@@ -106,6 +105,7 @@ WT_FILELIST = ['src/block/block_addr.c',
  'src/history/hs_cursor.c',
  'src/history/hs_rec.c',
  'src/history/hs_verify.c',
+ 'src/log/cur_log.c',
```
Comment (sueloverso) [inline dist/filelist.bzl:108]: I think this should be renamed to `log/log_cursor.c`. That makes it consistent with `btree/bt_cursor.c`, `lsm/lsm_cursor.c` and `history/hs_cursor.c`.

Comment (bvpvamsikrishna) [reply dist/filelist.bzl:108]: Yes, will make the change.

**Review by ajmorton — APPROVED** [review]
LGTM! 
The comment in Jira talks about this being part of a longer term change to move other cursors (history store, meta) into their respective modules, and based on this PR being LGTMd I assume this is the path we're taking.

Can you make sure we have a note on this in the modularity playbook? This seems like an check and change for developers to perform.

**Comment by bvpvamsikrishna** [comment]
@ajmorton , I made the changes to move the cur_json.c from cursor/ to support as there is no json cursor, all the function of cur_json or just utility functions, with this change, I believe the dependency of log on cursor is improved.
Could you review again? :)

File: src/include/json.h [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ * See the file LICENSE for redistribution information.
+ */
+
+#pragma once
+
+/*
+ * WT_JSON --
+ *   Conversion between JSON formatted string and WT CONFIG ITEM.
+ */
+
```
Comment (ajmorton) [inline src/include/json.h:15]: nit: blank line that can be removed

**Review by ajmorton — APPROVED** [review]
LGTM 👍

**Pull request merged**